### ISSUE TITLE
SelectBox: Add aria-label for select, drop-down list (and clients)

### ISSUE THREAD

**Issue by cleidigh** [Microsoft/vscode#53821]
@AccessibilityTestingTeam-TCS 
@bpasero 

The selectBoxList for windows and Linux does not allow for
a aria-label for either the parent select or the drop-down list.

Related: #52297

Since clients may instantiate a selectBox indirectly through something
such as an action bar (e.g. Terminal), I think it should be done
as a new methods such as selectBox.ariaLabel().  This would label both
the select and the drop-down list.

Proposed client label names:

Debug configurations:  "Debug Launch Configurations"
Breakpoint widget:  "Breakpoint Type
Terminal list: "Terminals"  FYI: @Tyriar 
Output list:  "Output list"
New Settings Editor:  item title?  FYI: @roblourens

**Comment by bpasero** [comment]
@cleidigh the aria label for the entire dropdown could maybe be an option to set on the [`SelectActionItem`](https://github.com/Microsoft/vscode/blob/bbafa06a3d3db60bbd06d4d4cd582586f7126d93/src/vs/base/browser/ui/actionbar/actionbar.ts#L761)?

As for the aria label on individual options, do you think that is needed? I think currently the option is being read out by NVDA as long as the items use the proper ARIA roles (maybe check https://www.w3.org/TR/wai-aria-practices/examples/listbox/listbox-collapsible.html for inspiration).

**Comment by cleidigh** [comment]
@roblourens - I think I've done a crappy job explaining where we are/proposed path - my bad!

Currently there is no label for either the parent select(custom or native mode), nor is there a label
on the drop-down list (separate element obviously).  If we had the label option for the select box,
we can then use this for both the parent select and the list and they will effectively look like the same element.  We can also set the expanded attribute on the list addressing the original issue.

For the new settings configuration, you could pass the label based on the configuration each time
you create a new select box.

You are correct in that the current set up does read out the options as selected, just no readout for the control state/name.

**Comment by cleidigh** [comment]
@bpasero 

Yes,  I can add the ISelectBoxOptions interface to SelectActionItem constructor for both the label,
minBottomMargin and future additions.  Pass to the select box constructor, no problem.

I think your second comment was based on my lousy description, you are correct the labels on the options  work appropriately (will add "selected" issue separately for the list
